query-compiler/query-compiler-wasm/src/lib.rs

```diff
@@ -0,0 +1,21 @@
+#[cfg(not(target_arch = "wasm32"))]
+mod arch {
+    // This crate only works in a Wasm environment.
+    // This conditional compilation block is here to make commands like
+    // `cargo clippy --all-features` happy, as `clippy` doesn't support the
+    // `--exclude` option (see: https://github.com/rust-lang/rust-clippy/issues/9555).
+    //
+    // This crate can still be inspected by `clippy` via:
+    // `cargo clippy --all-features -p schema-engine-wasm --target wasm32-unknown-unknown`
+}
```

Comment:
I looked into this today, and I believe this shouldn't be necessary if we use a cargo feature instead of `target_arch` logic to conditionally apply the `tsify` derive macro to the structs `driver-adapters` crate exports.